### PR TITLE
sonar update

### PR DESCRIPTION
**Pull request by LucianoCaravajal**
# iOS SDK

## Extra Notes

### CODE REVIEW

**Review by javrudsky — APPROVED** [review]
LGTM!

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            -Dsonar.projectName=${{ github.event.repository.name }}
+            -Dsonar.projectKey=splitio_ios-client
+            -Dsonar.c.file.suffixes=-
+            -Dsonar.cpp.file.suffixes=-
+            -Dsonar.objc.file.suffixes=-
+            -Dsonar.links.ci="https://github.com/splitio/${{ github.event.repository.name }}/actions"
+            -Dsonar.links.scm="https://github.com/splitio/${{ github.event.repository.name }}"
+            -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
+            -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }}
+            -Dsonar.pullrequest.base=${{ github.event.pull_request.base.ref }}
```
Comment (melinaurruchua) [inline .github/workflows/sonar.yml:31]: end line is missing

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - master
+jobs:
+  build:
+    name: Sonar
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: SonarQube Scan
```
Comment (melinaurruchua) [inline .github/workflows/sonar.yml:14]: can you let a line between the different steps?

File: .github/workflows/sonar.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  build:
+    name: Sonar
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (melinaurruchua) [inline .github/workflows/sonar.yml:11]: can you add a name?

**Comment by Israphel** [comment]
without the build, you won't have coverage info

**Comment by Israphel** [comment]
sonar has to run for PRs and merges (with different parameters)

**Comment by Israphel** [comment]
why not running everything in the same workflow if you have to build anyway?

**Review by Israphel — CHANGES_REQUESTED** [review]
sonarqube requires two kinds of executions with differente parameters.

if the build will be repeated, then combine the workflows.

**Pull request merged**